servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -522,6 +524,36 @@ void convertToLowerCase(Object obj) {
                     domainMeta.setSignAlgorithm(domainMeta.getSignAlgorithm().toLowerCase());
                 }
             }
+        },
+        ASSERTION_CONDITIONS {
+            @Override
+            void convertToLowerCase(Object obj) {
+                AssertionConditions assertionConditions = (AssertionConditions) obj;
+                // lower case key and value, assertion condition operator is an ENUM, hence left unchanged
+                assertionConditions.setConditionsList(assertionConditions.getConditionsList().stream()
+                        .map(c -> new AssertionCondition()
+                                .setConditionsMap(c.getConditionsMap().entrySet().stream()
+                                        .collect(Collectors.toMap(
+                                                e -> e.getKey().toLowerCase(),
+                                                e -> e.getValue().setValue(e.getValue().getValue().toLowerCase())
+                                        ))))
+                        .collect(Collectors.toList())
+                );
+            }
+        },
+        ASSERTION_CONDITION {
+            @Override
+            void convertToLowerCase(Object obj) {
+                AssertionCondition assertionCondition = (AssertionCondition) obj;
+                // lower case key and value, assertion condition operator is an ENUM, hence left unchanged
+                if (assertionCondition.getConditionsMap() != null) {
+                    assertionCondition.setConditionsMap(assertionCondition.getConditionsMap().entrySet().stream()
+                            .collect(Collectors.toMap(
```

Comment:
we should not create a new map. we can just iterate through and update values in place. let's not create new objects if not necessary